geocoding_ban/custom-recipes/geocoding_ban/recipe.json

```diff
@@ -0,0 +1,98 @@
+{
+    // Meta data for display purposes
+    "meta" : {
+        "label" : "Geocode with BAN",
+        "author" : "Ministère de l’intérieur — EIG 2017",
+        "tags" : ["geocoding", "dev"]
+    },
+
+    "kind" : "PYTHON",
+
+    "inputRoles" : [{
+        "name": "input",
+        "label": "Dataset to geocode",
+        "description": "what input A means",
+        "arity": "UNARY",
+        "required": true,
+        "acceptsDataset": true
+    }],
+    "outputRoles" : [{
+        "name": "output",
+        "label": "Output dataset",
+        "description": "what main output means",
+        "arity": "UNARY",
+        "required": true,
+        "acceptsDataset": true
+    }],
+
+    "params": [
+        {
+            "name": "server_address",
+            "label" : "Server hosting Addok",
+            "type": "STRING",
+            "description":"Full url with http://. Do not include the path /search",
+            "mandatory" : true,
+            "defaultValue" : "http://api-adresse.data.gouv.fr"
+        },
+        {
+            "name": "columns",
+            "label" : "Address columns",
+            "type": "COLUMNS",
+            "description":"Multiple columns will be concatenated",
```

Comment:
Add "columnRole": "input" (name of an input role), and you'll have autocompletion on the column names. Same for postcode and citycode.